pkg/logs/internal/tailers/file/rotate_nix.go

```diff
@@ -37,8 +38,19 @@ func (t *Tailer) DidRotate() (bool, error) {
 		return true, nil
 	}
 
+	lastReadOffset := t.lastReadOffset.Load()
+	fileOneSize := fi1.Size()
+
 	recreated := !os.SameFile(fi1, fi2)
-	truncated := fi1.Size() < t.lastReadOffset.Load()
+	truncated := fileOneSize < lastReadOffset
+
+	if recreated {
+		log.Debugf("File rotation detected due to recreation, f1: %v, f2: %v", fi1, fi2)
```

Comment:
How one would use FileInfo dumps here? I took one from the go playground, and it looks like this:
```
{passwd 340 420 {0 63750915359 0x53d960} {8 11 1 33188 0 0 0 0 340 4096 1 {1666168356 901477820} {1615318559 0} {1647438430 659553144} [0 0 0]}}
```

Reply:
Yes its a bit cryptic without field names isn't it. Changing to `%+v` which gives:
```
File stat: &{name:passwd size:340 mode:420 modTime:{wall:0 ext:63750915359 loc:0x53f960} sys:{Dev:8 Ino:11 Nlink:1 Mode:33188 Uid:0 Gid:0 X__pad0:0 Rdev:0 Size:340 Blksize:4096 Blocks:1 Atim:{Sec:1666204526 Nsec:899572849} Mtim:{Sec:1615318559 Nsec:0} Ctim:{Sec:1647438422 Nsec:453058250} X__unused:[0 0 0]}}
```

This is more to just have more data when debugging rather than looking for anything in particular, which is why I'm printing the whole `FileInfo` object (and why its at `DEBUG` log level)

For example, in a recent investigation I was curious if the "new" file had potentially been moved to a new `device` (ie, the log directory had been remounted on a different device). There was no way to get that information, hence this PR.